playground/frontend/playground_components/lib/widgets/split_view/controller.dart

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+import '../../constants/sizes.dart';
+
+class SplitViewController extends ChangeNotifier {
+  final bool isHorizontal;
+
+  SplitViewController({
+    required this.isHorizontal,
+  });
+
+  static const _minRatio = 0.3;
+  static const _maxRatio = 0.7;
+  static const _defaultRatio = 0.5;
+
+  double _ratio = _defaultRatio;
+  double _maxSize = 0;
```

Comment:
This size makes this controller aware of the surroundings which is an undesired coupling. It is sufficient to just store and return the ratio. So no `firstSize` and `secondSize` too.

Reply:
Changed to use https://pub.dev/packages/split_view